engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerSubprocessEventTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.timer;
+
+import static org.assertj.core.api.Assertions.assertThat;import io.camunda.zeebe.engine.util.EngineRule;import io.camunda.zeebe.model.bpmn.Bpmn;import io.camunda.zeebe.model.bpmn.BpmnModelInstance;import io.camunda.zeebe.protocol.record.Assertions;import io.camunda.zeebe.protocol.record.Record;import io.camunda.zeebe.protocol.record.intent.TimerIntent;import io.camunda.zeebe.protocol.record.value.TimerRecordValue;import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import org.junit.BeforeClass;import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class TimerSubprocessEventTest {
```

Comment:
❌ Nice job on the tests 👍 I think we can do this in a simpler way though. We can add a parameter to the [InterruptingEventSubprocessTest)[https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessTest.java] which uses a date time to trigger the timer. This class already has the testcases to verify the cases you have here.

It would be great to also put it in the [NonInterruptingEventSubprocessTest](https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/NonInterruptingEventSubprocessTest.java), but there is two tests that verify the timer is triggered twice, which it won't do when it's supposed to trigger on a specific date time. Maybe you can add an option to the parameters to skip these tests for this case?

Reply:
Makes a lot of sense, I was looking for a place to put them, but it's so much I must have missed it.